lib/Sema/MiscDiagnostics.cpp

```diff
@@ -2871,13 +2871,31 @@ static void checkStmtConditionTrailingClosure(TypeChecker &TC, const Expr *E) {
     DiagnoseWalker(TypeChecker &tc) : TC(tc) { }
 
     std::pair<bool, Expr *> walkToExprPre(Expr *E) override {
-      // Dig into implicit expression.
-      if (E->isImplicit()) return { true, E };
-      // Diagnose call expression.
-      if (auto CE = dyn_cast<CallExpr>(E))
+      if (E->isImplicit())
+        return { true, E };
+
+      switch (E->getKind()) {
+      case ExprKind::Closure:
+      case ExprKind::Paren:
+      case ExprKind::Tuple:
+      case ExprKind::TupleShuffle:
+      case ExprKind::ObjectLiteral:
+      case ExprKind::Array:
+      case ExprKind::Dictionary:
+        // Don't dig into expression which syntactically enclose other
+        // expressinos. e.g. [...], (...), and { ... }
```

Comment:
Typo: `expressinos`

Reply:
I could really go for an expressino right now.